airbytehq/airbyte
# convert container-orchestrator to micronaut

## What
- fix #13271

## How
- remove unused dependencies
  - apache-commons
  - jetty
  - config-persistence
  - db-lib
  - protocol-models
  - job-persistence
- add additional `deserialize` methods to the `Jsons` helper class
  - exposes more methods of the underlying jackson object-mapper
    - primarily for avoiding _unchecked casts_ issues
- rename `ContainerOrchestratorApp` to `Application`
  - remove busy-wait on `SUCCESS_FILE_NAME`
    - replace with k8s initContainer (#19088)
  - move logging initialization to EventListener classes
  - move dependency creation to Factory classes
- remove `WorkerHeartbeatServer` with its `HttpServlet`
  - replace with Micronaut `HeartbeatController` 
- combine `AsyncStateManager` interface with `DefaultAsyncStateManager` class into `AsyncStateManager` class 
- move `JobOrchestrator` classes to a new package
  - replace file reading methods of `JobOrchestrator` interface with injected dependencies to the implementations 
- add `application.yml`
- add `micronaut-banner.txt`
- add unit-tests
  - most of existing code did not contain any unit-tests as the code wasn't easily testable, some improvements have been made here and a lot of the new code now has tests   
- general code clean up
  - change visibility of methods to better match their expected usage (mostly moving them from the public scope)
  - remove unnecessary lambda usage (mostly moving to method references)
  - `final _type_` -> `final var`

## Oddities
- the `Application` class does a little too much currently and should probably be split into two different classes
- the `EventListeners.setEnvVars` method should probably be replaced with the env-vars being set on the container itself (in the worker when it creates this pod and containers)
- the `@Secured` annotation on the `HeartbeatController` could (should?) be replaced with something else
